packages/curve-ui-kit/src/features/switch-chain/ui/ChainSwitcher.tsx

```diff
@@ -42,11 +45,10 @@ export const ChainSwitcher = ({ networks, chainId }: ChainSwitcherProps) => {
 
   const onClick = options.length > 1 ? toggle : openSnackbar
   const top = useLayoutStore((state) => state.navHeight)
-  const network: NetworkDef | undefined = networks[chainId] // the network may not be defined when switching apps
   return (
     <>
       <IconButton size="small" onClick={onClick} data-testid="btn-change-chain">
-        {network && <ChainSwitcherIcon network={network} />}
+        {networkId && <ChainSwitcherIcon networkId={networkId} />}
```

Comment:
Is there a case where this is false, now that it's retrieved from the URL which always has a network defined?

Reply:
If you navigate to the root it can be undefined